src/main/scala/analysis/InterprocSteensgaardAnalysis.scala

```diff
@@ -0,0 +1,381 @@
+                    val X1 = localAssign.lhs
+                    val X2 = variable
+                    unify(IdentifierVariable(RegisterVariableWrapper(X1)), IdentifierVariable(RegisterVariableWrapper(X2)))
+                  case _ => // do nothing
+                }
+            }
+          case memoryAssign: MemoryAssign =>
+            // *X1 = X2: [[X1]] = ↑a ^ [[X2]] = a where a is a fresh term variable
+            val X1_star = exprToRegion(memoryAssign.rhs.index, cmd)
+            val X2 = evaluateExpressionWithSSA(memoryAssign.rhs.value, constantProp(n))
+            var possibleRegions = Set[MemoryRegion]()
+            if X2.isEmpty then
+              Logger.debug("Maybe a region: " + exprToRegion(memoryAssign.rhs.value, cmd))
+              possibleRegions = exprToRegion(memoryAssign.rhs.value, cmd)
+            Logger.debug("X2 is: " + X2)
+            Logger.debug("Evaluated: " + memoryAssign.rhs.value)
+            Logger.debug("Region " + X1_star)
+            Logger.debug("Index " + memoryAssign.rhs.index)
+            val alpha = FreshVariable()
+            X1_star.foreach(
+              x =>
+                unify(ExpressionVariable(x), PointerRef(alpha))
+                x.content.addAll(X2)
+                x.content.addAll(possibleRegions.filter(r => r != x))
+            )
+            X2.foreach(
+              x => unify(alpha, ExpressionVariable(x))
+            )
+            possibleRegions.foreach(
+              x => unify(alpha, ExpressionVariable(x))
+            )
+          case _ => // do nothing TODO: Maybe LocalVar too?
+        }
+      case _ => // do nothing
+    }
+  }
+
+  private def unify(t1: Term[StTerm], t2: Term[StTerm]): Unit = {
+    //Logger.info(s"univfying constraint $t1 = $t2\n")
+    solver.unify(
+      t1,
+      t2
+    ) // note that unification cannot fail, because there is only one kind of term constructor and no constants
+  }
+
+  /** @inheritdoc
+    */
+  def pointsTo(): Map[CfgNode, Set[RegisterVariableWrapper | MemoryRegion]] = {
+    val solution = solver.solution()
+    val unifications = solver.unifications()
+    Logger.debug(s"Solution: \n${solution.mkString(",\n")}\n")
+    Logger.debug(s"Sets: \n${unifications.values
+      .map { s =>
+        s"{ ${s.mkString(",")} }"
+      }
+      .mkString(", ")}")
+
+    val vars = solution.keys.collect { case id: IdentifierVariable => id }
+    val pointsto = vars.foldLeft(Map[Object, Set[RegisterVariableWrapper | MemoryRegion]]()) { case (a, v: IdentifierVariable) =>
+      val pt = unifications(solution(v))
+        .collect({
+          case PointerRef(IdentifierVariable(id)) => id
+          case PointerRef(AllocVariable(alloc))   => alloc
+        })
+        .toSet
+      a + (v.id -> pt)
+    }
+    Logger.debug(s"\nPoints-to:\n${pointsto.map(p => s"${p._1} -> { ${p._2.mkString(",")} }").mkString("\n")}\n")
+    pointsto.asInstanceOf[Map[CfgNode, Set[RegisterVariableWrapper | MemoryRegion]]]
```

Comment:
pointsto is never actually a Map[CfgNode, _] it is a Map[RegisterVariableWrapper, _] but is incorrectly. Scala doesn't enforce these sorts of collection parameter types at runtime so it all goes through but is incredibly confusing to try to understand.